src/main/java/org/zalando/nakadi/service/EventTypeService.java

```diff
@@ -123,17 +139,37 @@ public EventTypeService(final EventTypeRepository eventTypeRepository, final Top
             eventTypeRepository.removeEventType(eventTypeName);
             topicRepository.deleteTopic(eventType.get().getTopic());
             return Result.ok();
+        } catch (final InterruptedException e) {
+            Thread.currentThread().interrupt();
+            LOG.error("Failed to wait for timeline switch", e);
+            return Result.problem(Problem.valueOf(Response.Status.SERVICE_UNAVAILABLE,
+                    "Event type is currently in maintenance, please repeat request"));
+        } catch (final TimeoutException e) {
+            LOG.error("Failed to wait for timeline switch", e);
+            return Result.problem(Problem.valueOf(Response.Status.SERVICE_UNAVAILABLE,
+                    "Event type is currently in maintenance, please repeat request"));
         } catch (final TopicDeletionException e) {
             LOG.error("Problem deleting kafka topic " + eventTypeName, e);
             return Result.problem(e.asProblem());
         } catch (final NakadiException e) {
             LOG.error("Error deleting event type " + eventTypeName, e);
             return Result.problem(e.asProblem());
+        } finally {
+            try {
+                if (deletionCloser != null) {
+                    deletionCloser.close();
+                }
+            } catch (final IOException e) {
+                LOG.error("Exception occurred when releasing usage of event-type", e);
+            }
         }
     }
 
     public Result<Void> update(final String eventTypeName, final EventTypeBase eventTypeBase, final Client client) {
+        Closeable updatingCloser = null;
         try {
+            updatingCloser = timelineSync.workWithEventType(eventTypeName, nakadiSettings.getTimelineWaitTimeoutMs());
```

Comment:
please see this comment from Ricardo https://github.com/zalando/nakadi/pull/548#issuecomment-276060597

The possible case can be the change of retention time (which will work sooner or later)